experimental/dds/tree2/src/test/undo/undoRedoManager.spec.ts

```diff
@@ -45,42 +59,105 @@ describe("UndoRedoManager", () => {
 			manager.trackCommit(fakeInvertedCommit, UndoRedoManagerCommitType.Undo);
 			// The head undo commit will not be the new inverted commit
 			assert.equal(manager.headUndoable, undefined);
+			// The fake undo commit will now be redoable
+			assert.equal(manager.headRedoable?.commit, fakeInvertedCommit);
 		});
-	});
 
-	describe("undo", () => {
-		it("should undo the head undo commit", () => {
+		it("pops the head redoable commit and pushes to the undoable commit tree when a redo commit is tracked", () => {
 			const initialCommit = {
 				commit: createTestGraphCommit([], 0, localSessionId),
 				repairData: new MockRepairDataStore(),
 			};
 			const manager = undoRedoManagerFactory(initialCommit);
-			const undoableCommit = createTestGraphCommit([0], 1, localSessionId);
-			manager.trackCommit(undoableCommit, UndoRedoManagerCommitType.Undoable);
-			manager.undo();
 			const fakeInvertedCommit = createTestGraphCommit([0, 1], 2, localSessionId);
-			manager.trackCommit(fakeInvertedCommit, UndoRedoManagerCommitType.Undo);
+			manager.trackCommit(fakeInvertedCommit, UndoRedoManagerCommitType.Redo);
+			// The head undo commit will not be the new inverted commit
+			assert.equal(manager.headRedoable, undefined);
+			// The fake undo commit will now be redoable
+			assert.equal(manager.headUndoable?.commit, fakeInvertedCommit);
+		});
+
+		it("should add undoable commits to the undoable commit tree without changing the redoable commit tree", () => {
+			const initialCommit = {
+				commit: createTestGraphCommit([], 0, localSessionId),
+				repairData: new MockRepairDataStore(),
+			};
+			const manager = undoRedoManagerFactory(initialCommit);
+			const fakeInvertedCommit = createTestGraphCommit([0, 1], 2, localSessionId);
+			manager.trackCommit(fakeInvertedCommit, UndoRedoManagerCommitType.Undoable);
+			// The head undo commit will not be the new inverted commit
+			assert.equal(manager.headUndoable?.commit, fakeInvertedCommit);
+			// The fake undo commit will now be redoable
+			assert.equal(manager.headRedoable, undefined);
+		});
+
+		it("should add redoable commits to the redoable commit tree without changing the undoable commit tree", () => {
+			const initialCommit = {
+				commit: createTestGraphCommit([], 0, localSessionId),
+				repairData: new MockRepairDataStore(),
+			};
+			const manager = undoRedoManagerFactory(initialCommit);
+			const fakeInvertedCommit = createTestGraphCommit([0, 1], 2, localSessionId);
+			manager.trackCommit(fakeInvertedCommit, UndoRedoManagerCommitType.Redoable);
+			// The head undo commit will not be the new inverted commit
 			assert.equal(manager.headUndoable, initialCommit);
+			// The fake undo commit will now be redoable
+			assert.equal(manager.headRedoable?.commit, fakeInvertedCommit);
+		});
+	});
+
+	describe("undo", () => {
```

Comment:
```suggestion
	describe("undo/redo", () => {
```

Or split into two describe blocks